### PR TITLE
Fix stalling on upgrade from latest minor release with flux

### PR DESCRIPTION
**Pull request by cxbrowne1207**
*Issue #, if available:*

*Description of changes:*
This PR adds a `ClusterSpecGenerate` struct when marshalling the cluster spec. We needed this specifically such that we could generate a yaml with bundlesRef null so that it would
not be omitted from the kubeapi-server.

There was an issue when upgrading from the latest minor release with gitps v0.16.2 to v0.17 during the "Applying eksa yaml resources to cluster" step stalled indefinitely because both the newly added eksaVersion field and the bundlesRef were populated in submission to the kubeapi server. There is a webhook validation that does not allow both to be set at the same time, so that was throwing an error in the kube-api-server pod. This is strange because, in the original submission of the Cluster resource yaml, bundlesRef was nil. As a result of using the force apply for this operation, it was attempting to delete the eksa Cluster resource and stalled on that operation.

This solution is a temporary patch for the issue. After addressing, we should clean this up.

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by cxbrowne1207** [comment]
/hold 

Running a few e2e tests before merging

File: pkg/api/v1alpha1/cluster_types.go
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster_types.go:1448]: Can we add a quick test for this function? It should be simple.

**Review by vivek-koppuru — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**
